src/lib/core/Program.ts

```diff
@@ -123,18 +132,7 @@ export class Program {
       depthWrite = true,
       depthFunc = gl.LESS,
       transformFeedbackVaryings
-    }: {
-      vertex?: string
-      fragment?: string
-      uniforms?: WTCGLUniformArray
-      transparent?: boolean
-      cullFace?: GLenum
-      frontFace?: GLenum
-      depthTest?: boolean
-      depthWrite?: boolean
-      depthFunc?: GLenum
-      transformFeedbackVaryings?: string[]
-    } = {}
+    }: ProgramOptions
```

Comment:
On line `164`, `shaderSource` requires a vertex source, so this can't be empty. It's why I didn't mark it as `Partial` either. Same for the fragment on line `178`. I could make these optional and check beforehand though